c7n/filters/vpc.py

```diff
@@ -166,7 +166,7 @@ def filter_ignored(self, resources):
 
     def process_resource(self, r, resource_sgs, resource_subnets, key):
         evaluation = []
-        if 'subnet' in self.compare:
+        if 'subnet' in self.compare and len(resource_subnets):
```

Comment:
since you're doing the same comparison (`len(resource_subnets)`) over and over again, you should pull it out to the top level before the first if statement

Reply:
it's two different comparisons, `len(resource_sqs)` and `len(resource_subnets)`.  and while there are two occurrences of each, each check is in a different context (`...LocationAbsent` or `...LocationCardinality` vs. `ResourceLocationMismatch`).

Reply:
Gotcha, the way that the diff showed up made me think it was all in one big block, the only thing I would change then is to remove the `len` call, since you can just do `if 'subnet' in self.compare and resource_subnets:`

Reply:
update submitted.